aws_lambda_powertools/utilities/trigger/sns_event.py

```diff
@@ -0,0 +1,123 @@
+        return self["Type"]
+
+    @property
+    def value(self) -> str:
+        """The user-specified message attribute value."""
+        return self["Value"]
+
+
+class SNSMessage(DictWrapper):
+    @property
+    def signature_version(self) -> str:
+        """Version of the Amazon SNS signature used."""
+        return self["Sns"]["SignatureVersion"]
+
+    @property
+    def timestamp(self) -> str:
+        """The time (GMT) when the subscription confirmation was sent."""
+        return self["Sns"]["Timestamp"]
+
+    @property
+    def signature(self) -> str:
+        """Base64-encoded "SHA1withRSA" signature of the Message, MessageId, Type, Timestamp, and TopicArn values."""
+        return self["Sns"]["Signature"]
+
+    @property
+    def signing_cert_url(self) -> str:
+        """The URL to the certificate that was used to sign the message."""
+        return self["Sns"]["SigningCertUrl"]
+
+    @property
+    def message_id(self) -> str:
+        """A Universally Unique Identifier, unique for each message published.
+
+        For a message that Amazon SNS resends during a retry, the message ID of the original message is used."""
+        return self["Sns"]["MessageId"]
+
+    @property
+    def message(self) -> str:
+        """A string that describes the message. """
+        return self["Sns"]["Message"]
+
+    @property
+    def message_attributes(self) -> Dict[str, SNSMessageAttribute]:
+        return {k: SNSMessageAttribute(v) for (k, v) in self["Sns"]["MessageAttributes"].items()}
+
+    @property
+    def get_type(self) -> str:
+        """The type of message.
+
+        For a subscription confirmation, the type is SubscriptionConfirmation."""
+        # Note: this name conflicts with existing python builtins
+        return self["Sns"]["Type"]
+
+    @property
+    def unsubscribe_url(self) -> str:
+        """A URL that you can use to unsubscribe the endpoint from this topic.
+
+        If you visit this URL, Amazon SNS unsubscribes the endpoint and stops sending notifications to this endpoint."""
+        return self["Sns"]["UnsubscribeUrl"]
+
+    @property
+    def topic_arn(self) -> str:
+        """The Amazon Resource Name (ARN) for the topic that this endpoint is subscribed to."""
+        return self["Sns"]["TopicArn"]
+
+    @property
+    def subject(self) -> str:
+        """The Subject parameter specified when the notification was published to the topic."""
+        return self["Sns"]["Subject"]
+
+
+class SNSEventRecord(DictWrapper):
+    @property
+    def event_version(self) -> str:
+        """Event version"""
+        return self["EventVersion"]
+
+    @property
+    def event_subscription_arn(self) -> str:
+        return self["EventSubscriptionArn"]
+
+    @property
+    def event_source(self) -> str:
+        """The AWS service from which the SNS event record originated. For SNS, this is aws:sns"""
+        return self["EventSource"]
+
+    @property
+    def sns(self) -> SNSMessage:
```

Comment:
> ```suggestion
> 
>     def message(self) -> SNSMessage:
> 
> ```
> 
> 
> 
> Can we call this "message" instead?

@cakepietoast - sure I can do this. I tended to keep the name based on the json schema. 

1. Lower case
2. Snake case
3. Add `get`_ when it conflicts with a python built in. 

The reason was in case we end up generating these from an official schema. 


I can make this change so long.

Reply:
I just had another look at the event json and I changed my mind. Lets leave this as it is, sorry!